src/coreclr/src/gc/gc.cpp

```diff
@@ -15638,6 +15643,7 @@ int gc_heap::joined_generation_to_condemn (BOOL should_evaluate_elevation,
 
         if (full_compact_gc_p)
         {
+            gc_heap::gc_data_global.gen_to_condemn_reasons.set_condition(j3);
```

Comment:
>gc_heap::gc_data_global.gen_to_condemn_reasons.set_condition(j3); [](start = 12, length = 65)

it'd be nicer to record separate reasons for when we set full_compact_gc_p to true